File: backend/src/main/java/com/example/backend/service/ReactService.java
package com.example.backend.service;

import com.example.backend.model.request.ReactRequest;
import com.example.backend.model.respon.ReactResponse;

public interface ReactService {
    // React hoặc hủy react trên bài viết/bình luận (toggle reaction)
    ReactResponse reactPostOrComment(Long userId, ReactRequest request);

    // Lấy danh sách react của một bài viết
    java.util.List<ReactResponse> getReactionsByPostId(Long postId);

    // Lấy danh sách react của một bình luận
    java.util.List<ReactResponse> getReactionsByCommentId(Long commentId);
}

File: frontend/src/service/impl/ReactService.ts
import { http } from "../http/http";
import type { ReactRequest } from "../../utils/RequestType";
import type { ApiDataResponse, ReactResponse } from "../../utils/ResponType";

export const reactPostOrComment = (currentUserId: number, request: ReactRequest) => {
    return http.post<ApiDataResponse<ReactResponse>>(`/reacts?currentUserId=${currentUserId}`, request);
};

export const getReactionsByPostId = (postId: number) => {
    return http.get<ApiDataResponse<ReactResponse[]>>(`/reacts/posts/${postId}`);
};

export const getReactionsByCommentId = (commentId: number) => {
    return http.get<ApiDataResponse<ReactResponse[]>>(`/reacts/comments/${commentId}`);
};
